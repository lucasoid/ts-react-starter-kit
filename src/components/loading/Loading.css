span.loadingText {
    margin-bottom: 1px;
    height: 1em;
    width: 100%;
    display: inline-block;
    content: ' ';
}

span.loadingText.light {
    background: #dfdfdf;
}

span.loadingText.dark {
    background: #888888;
}
