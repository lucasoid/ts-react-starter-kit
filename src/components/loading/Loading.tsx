import * as React from 'react';
import { ThemeContext, Themes } from '~theme';
import * as styles from './Loading.css';

interface ILoadingProps {
    el: string;
    lines: number;
    width: string;
    children?: any;
}

export const Loading: React.FC<ILoadingProps> = props => {
    let { el, lines, width } = props;
    let display = [];
    const { theme } = React.useContext(ThemeContext);

    for (let i = 0; i < lines; i++) {
        display.push(
            React.createElement(
                el,
                { key: i, style: { width: width } },
                <span className={`${styles.loadingText} ${theme === Themes.DARK ? styles.dark : styles.light}`} />,
            ),
        );
    }
    return <>{display}</>;
};
