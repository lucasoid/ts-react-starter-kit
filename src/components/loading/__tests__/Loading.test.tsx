import * as React from 'react';
import { shallow } from 'enzyme';
import { Loading } from '../Loading';
import { Themes } from '~theme';

jest.mock('react', () => {
    const original = jest.requireActual('react');
    return {
        __esModule: true,
        ...original,
        useContext: jest.fn(),
    };
});

describe('Loading', () => {
    it('displays a themed placeholder UI', () => {
        // @ts-ignore
        React.useContext.mockImplementationOnce(() => ({
            theme: Themes.LIGHT,
        }));
        let lightWrapper = shallow(<Loading width={'50%'} lines={3} el={'p'} />);
        expect(lightWrapper).toMatchSnapshot();

        // @ts-ignore
        React.useContext.mockImplementationOnce(() => ({
            theme: Themes.DARK,
        }));
        let darkWrapper = shallow(<Loading width={'100px'} lines={1} el={'h1'} />);
        expect(darkWrapper).toMatchSnapshot();
    });
});
