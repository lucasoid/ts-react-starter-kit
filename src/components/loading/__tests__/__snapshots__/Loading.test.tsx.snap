// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading displays a themed placeholder UI 1`] = `
<Fragment>
  <p
    key="0"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <span
      className="loadingText light"
    />
  </p>
  <p
    key="1"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <span
      className="loadingText light"
    />
  </p>
  <p
    key="2"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <span
      className="loadingText light"
    />
  </p>
</Fragment>
`;

exports[`Loading displays a themed placeholder UI 2`] = `
<Fragment>
  <h1
    key="0"
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <span
      className="loadingText dark"
    />
  </h1>
</Fragment>
`;
